text/0000-localizing-api-messages.md

```diff
@@ -0,0 +1,135 @@
+
+- Start Date: 03-29-2022
+- RFC PR: 
+- FOLIO Issue: 
+- Submitter : Radhakrishnan Gopalakrishnan (rgopalakrishnan@ebsco.com)
+- Co-Submitter(s) : Zak Burke (zburke@ebsco.com)
+- Sub Group :
+
+# Localizing API (Backend) Messages
+
+## Summary
+  The purpose of this RFC is to define an approach that allows API developer to provide
+messages in the API response depending on the language preference set in the request. 
+
+
+## Motivation
+
+- Help users understand the information coming from the application backend using a language they understand better
+- To allow FOLIO platform to be used by users across the world 
+- To create a tremendous opportunity for growth that we can never could have achieved within just one country.
+
+### Terminology
+- Static Message - E.g. _This is something you need translate_
+- Static Message with placeholder - _You can ONLY renew {n} times_
+ 
+
+### In Scope Requirements/Use cases
+- Return localized messages based on the value passed in the accept-language header
+- Handle static messages, static messages with placeholders
+
+### Out of Scope Requirements/Use cases
+- [Plural Syntax](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Pluralsyntax)
+- Naming convention to be used for keys
+- Process for managing (decoupled from lokalise.com) translations
+- Returning formatted (HTML/Markdown) messages
+- Usage of [soft hyphen](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Softhyphentobreakwords) to break messages
+- Support customization per tenant
+- Process for back porting API/Backend messages similar to what we have for the front end 
+- Support for controlled vocabulary (runtime data/data coming from DB tables. E.g. Patron Groups for tenants)
+
+## Detailed Explanation/Design
+- Locale specific translations MUST be stored in a properties file
+- Properties file naming convention
+  - default locale (en_US) - somename.properties
+  - specific locale - somename_fr.properties (For french),  somename_de.properties (For german)
+- Properties file MUST BE loaded using the classpath. AVOID using file locations
+- Properties file entries WILL use the standard key value pair format
+- When using application frameworks (spring, vertx, etc.) implement localization using the framework specific implementation (if there is one)
+- When handling messages with placeholders, replace the placeholders with the actual value on the server side before
+  sending it back to the client
+- STOP using the lang parameter.
+- accept-language header value MUST be in ll_CC format. , where ll is a two-letter language code, 
+  and CC is a two-letter country code.
+- When accept-language header is missing or has an incorrect value, return the message in en_US (or the default locale configured)
+
+
+## Rationale and Alternatives
```

Comment:
Thank you for collecting and writing up this "Rationale and Alternatives" section.
We already have made a decision whether messages should be generated by the back-end module itself or at some other module: https://wiki.folio.org/display/DD/Handle+i18n+where+messages+are+generated
Therefore this decision is out-of-scope of this RFC.
This section "Rationale and Alternatives" belongs to that decision. I've copied this section over there: https://wiki.folio.org/display/DD/Handle+i18n+where+messages+are+generated
Can this section removed from this RFC?